src/ir/module-utils.cpp

```diff
@@ -46,6 +46,15 @@ Function* copyFunction(Function* func,
                        Module& out,
                        Name newName,
                        std::optional<std::vector<Index>> fileIndexMap) {
+  auto ret = copyFunctionWithoutAdd(func, out, newName, fileIndexMap);
+  return out.addFunction(std::move(ret));
+}
+
+std::unique_ptr<Function>
+copyFunctionWithoutAdd(Function* func,
```

Comment:
A downside to `copyAndAddFunction` (or to keeping the name but dropping the add) is that then it would not be consistent with `copyGlobal, copyTable, etc.` which all do the natural thing and add.

It is the not-adding that is the slightly odd thing here, but this code really benefits from it, and it seems not that big a burden to have a public API for it I think.

Reply:
We definitely want consistency. IMO it would be cleanest if none of the `copyXXX` functions did the add because then they would all be pure, but no need to block this PR on that.